drivers/serial/Kconfig.ns16550

```diff
@@ -6,6 +6,8 @@ menuconfig UART_NS16550
 	depends on DT_HAS_NS16550_ENABLED
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
+	select SERIAL_SUPPORT_ASYNC if BOARD_INTEL_RPL_S_CRB
```

Comment:
No. A board specific configuration needs to be in the board directory, and should not be done at the driver level.

Also, the actual implementation requires LPSS DMA to work. So put `select SERIAL_SUPPORT_ASYNC` under `UART_NS16550_LPSS_DMA`.